### DISCUSSION TITLE
Partial Prerendering & CDN Support

### DISCUSSION THREAD

**Discussion by psimk** [vercel/next.js#74221] [App Router] [answered]
Hi,

In the [Patrial Prerendering deep div](https://youtu.be/zL1f0SJmzt0) @wyattjoh hinted that Vercel will be opening up Partial Prerendering as a capability to other CDNs. This sounds incredibly exciting for my company, as due to corporate policy, we run everything on-prem. Plus from an educational standpoint it's interesting to see how static + dynamic requests could be combined in a production setting.

- Is there a public roadmap available for PPR in CDNs?
- Can anything be shared about how the Vercel CDN currently implements single-request PPR?

**Comment by wyattjoh** [comment, marked as answer]
We're currently working towards stabilizing PPR for self hosted (without a CDN) at the moment, but we don't currently have anything to share at this time as to how PPR will function without the Vercel CDN today. Stay tuned!

As for the single-request PPR implementation, the self-hosted version of Next.js does this today! The only difference on Vercel is that our CDN performs the "serve static file and then stream on dynamic response" part of it at the edge.

**Reply by sutho91** [reply]
Is there another discussion or issue to follow to get updates on how / when self hosted ppr + CDN will happen/work? Ppr sounds great, but if we can't use a CDN then a lot of the benefits might not be achieved?

**Reply by tsoos99dev** [reply]
@wyattjoh Do you have any updates regarding this? I'd be interested in setting up PPR with AWS Cloudfront.
